src/test_util/mod.rs

```diff
@@ -289,6 +293,34 @@ pub fn random_events_with_stream(
     (events, stream)
 }
 
+pub fn metrics_with_stream(
+    count: i32,
+    batch: Option<BatchNotifier>,
+) -> (Vec<Event>, impl Stream<Item = EventArray>) {
+    let events: Vec<_> = (0..count).map(create_event).collect();
+    let stream = map_batch_stream(
+        stream::iter(events.clone()).map(|event| event.into_log()),
+        batch,
+    );
```

Comment:
This will panic because the events we are passing through are metrics and not log messages. We can instead just call `map_event_batch_stream` which takes `Event`.
```suggestion
    let stream = map_event_batch_stream(stream::iter(events.clone()), batch);
```